packages/expo-updates/android/src/androidTest/java/expo/modules/updates/launcher/SelectionPolicyFilterAwareTest.java

```diff
@@ -0,0 +1,147 @@
+    List<UpdateEntity> updatesToDelete = selectionPolicy.selectUpdatesToDelete(Arrays.asList(updateDefault1, updateRollout1, updateDefault2, updateRollout2), updateRollout2, manifestFilters);
+
+    Assert.assertEquals(2, updatesToDelete.size());
+    Assert.assertTrue(updatesToDelete.contains(updateDefault1));
+    Assert.assertFalse(updatesToDelete.contains(updateRollout1));
+    Assert.assertTrue(updatesToDelete.contains(updateDefault2));
+    Assert.assertFalse(updatesToDelete.contains(updateRollout2));
+  }
+
+  @Test
+  public void testSelectUpdatesToDelete_NoneOlderMatching() {
+    // if there is no older update that matches the manifest filters, just keep the next newest one
+    List<UpdateEntity> updatesToDelete = selectionPolicy.selectUpdatesToDelete(Arrays.asList(updateDefault1, updateDefault2, updateRollout2), updateRollout2, manifestFilters);
+
+    Assert.assertEquals(1, updatesToDelete.size());
+    Assert.assertTrue(updatesToDelete.contains(updateDefault1));
+    Assert.assertFalse(updatesToDelete.contains(updateDefault2));
+    Assert.assertFalse(updatesToDelete.contains(updateRollout2));
+  }
+
+  @Test
+  public void testShouldLoadNewUpdate_NormalCase_NewUpdate() {
+    boolean actual = selectionPolicy.shouldLoadNewUpdate(updateRollout2, updateRollout1, manifestFilters);
+    Assert.assertTrue(actual);
+  }
+
+  @Test
+  public void testShouldLoadNewUpdate_NormalCase_NoUpdate() {
+    boolean actual = selectionPolicy.shouldLoadNewUpdate(updateRollout1, updateRollout1, manifestFilters);
+    Assert.assertFalse(actual);
+  }
+
+  @Test
+  public void testShouldLoadNewUpdate_NoneMatchingFilters() {
+    // should choose to load an older update if the current update doesn't match the manifest filters
+    boolean actual = selectionPolicy.shouldLoadNewUpdate(updateRollout1, updateDefault2, manifestFilters);
+    Assert.assertTrue(actual);
+  }
+
+  @Test
+  public void testShouldLoadNewUpdate_NewerExists() {
+    boolean actual = selectionPolicy.shouldLoadNewUpdate(updateRollout1, updateRollout2, manifestFilters);
+    Assert.assertFalse(actual);
+  }
+
+  @Test
+  public void testShouldLoadNewUpdate_OlderExistsMatchingFilters() {
+    // should not choose to load a new update that doesn't match the manifest filters
+    // if there is an existing older update that matches the filters
+    boolean actual = selectionPolicy.shouldLoadNewUpdate(updateDefault2, updateRollout1, manifestFilters);
+    Assert.assertFalse(actual);
+  }
+
+  @Test
+  public void testShouldLoadNewUpdate_NoneExistsMatchingCurrentFilters() {
+    // should choose to load a new update that doesn't match the manifest filters
+    // if no existing updates match the manifest filters
+    boolean actual = selectionPolicy.shouldLoadNewUpdate(updateDefault2, null, manifestFilters);
+    Assert.assertTrue(actual);
+  }
+
+  @Test
+  public void testShouldLoadNewUpdate_NoneExistsMatchingNewFilters() {
```

Comment:
Do we want this behavior? What are its consequences? If the server sends down a manifest that doesn't match the filters, do we want the app to update to it?

Reply:
That's a good point -- discussed with @jkhales and we should return false here, essentially sanity checking that a manifest passes its own filters before loading it (and persisting the filters).